model/filters.go

```diff
@@ -59,6 +61,33 @@ type FilterPredicate struct {
 	Value     interface{} `json:"value" bson:"value"`
 }
 
+type GeoDistanceFilter struct {
+	GeoDistance GeoDistance `json:"geo_distance" bson:"geo_distance"`
+}
+
+type GeoDistance struct {
+	Distance string   `json:"distance" bson:"distance"`
+	Location GeoPoint `json:"location" bson:"location"`
+}
+
+type GeoBoundingBoxFilter struct {
+	GeoBoundingBox GeoBoundingBox `json:"geo_bounding_box" bson:"geo_bounding_box"`
+}
+
+type GeoBoundingBox struct {
+	Location BoundingBox `json:"location" bson:"location"`
+}
+
+type BoundingBox struct {
+	TopLeft     GeoPoint `json:"top_left" bson:"top_left"`
+	BottomRight GeoPoint `json:"bottom_right" bson:"bottom_right"`
+}
+
+type GeoPoint struct {
+	Lat float32 `json:"lat" bson:"lat"`
+	Lon float32 `json:"lon" bson:"lon"`
```

Comment:
another nit, but I am usre we can afford the full names in the identifiers.
```suggestion
	Latitude float32 `json:"lat" bson:"lat"`
	Longitude float32 `json:"lon" bson:"lon"`
```

Reply:
as you wish